Improved error logging.

mogren opened the pull request:
Some errors were not logged correctly or at all.

*Description of changes:*
Logging changes, and removed an unused return value.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

mattlandis commented:
I think the example you are pointing at is already fixed in master: https://github.com/aws/amazon-vpc-cni-k8s/pull/267

mogren commented:
Well, I added two missing errors that was never logged, and removed the returned error from `eniIPPoolReconcile()` since it only ever returned `nil` and was never checked. 

Only tested by building and running unit tests.

nckturner commented:
Formatting and error logging improvements, LGTM
